core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -529,4 +563,59 @@ public static void setCreationTime(final Message msg) {
         }
     }
 
+    /**
+     * Verify if a message is signalling that the
+     * device for which it was received should be ready to receive a command.
+     * This is evaluated at the point in time this method is invoked.
+     *
+     * @param msg Message that is evaluated.
+     * @return Boolean {@code true} if the message signals that the device now should be ready to receive a command, {@code false}
+     * otherwise.
+     * @throws NullPointerException If msg is {@code null}.
+     */
+    public static Boolean isDeviceCurrentlyReadyForCommands(final Message msg) {
+
+        return Optional.ofNullable(MessageHelper.getTimeUntilDisconnect(msg)).map(ttd -> {
+            if (ttd == MessageHelper.TTD_VALUE_ALWAYS_READY) {
+                return Boolean.TRUE;
+            } else if (ttd == 0) {
+                return Boolean.FALSE;
+            } else {
+                final Instant creationTime = Instant.ofEpochMilli(msg.getCreationTime());
+                return Instant.now().isBefore(creationTime.plusSeconds(ttd));
+            }
+        }).orElse(Boolean.FALSE);
+    }
+
+    /**
+     * Provide an instance of {@link TimeUntilDisconnectNotification} if a message indicates that the device sending it
+     * is currently ready to receive an upstream message.
+     * <p>
+     * If this is not the case, the returned {@link Optional} will be empty.
+     *
+     * @param msg Message that is evaluated.
+     * @return Optional containing an instance of the class {@link TimeUntilDisconnectNotification} if the device is considered
+     * being ready to receive an upstream message or is empty otherwise.
+     * @throws NullPointerException If msg is {@code null}.
+     */
+    public static Optional<TimeUntilDisconnectNotification> fromMessage(final Message msg) {
```

Comment:
I would expect a static factory method of this name to create an instance of `Constants` because it is not qualified with a name of the type of object it creates. Can't we simply move this method to `TimeUntilDisconnectNotification` or is there another dependency from `MessageHelper` on `TimeUntilDisconnectNotification`? Otherwise, can we at least rename this method to e.g. `newTtdNotification`?

Reply:
Looking at it again I fully see your concern - I think it is best moved to `TimeUntilDisconnectNotification` where it belongs to as a static factory method.
The dependency introduced for `TimeUntilDisconnectNotification` to `MessageHelper` I do not consider being critical.